static/js/operation.js

```diff
@@ -123,75 +123,76 @@ function parseAssembly(text, data, minContigLength) {
   var cols = ['id', 'length', 'gc', 'coverage']; // the information available in the contig titles
   var df = [];
 
-  var id = 0;
+  var id = NaN;
   var length = 0;
   var gc = 0;
   var coverage = 0;
-  var currentContigLength = 0;
-  var gcCount = 0;
 
   for (var i = 0; i < lines.length; i++) {
     var line = lines[i];
-    if (line.search(/^>{1}/gm) === 0) { // Checking if the current line is a contig title
-        [id, length, coverage] = parseContigTitles(line, format);
-        if (length <= minContigLength) 
-          continue;
-
-        // append dataframe with contig title information
-        df.push([id, length, , coverage]) // gc not calculated yet
-        gcCount = 0;
-        currentContigLength = 0;
+    if (line.charAt() === '>') { // Checking if the current line is a contig title
+        if (id !== NaN && length > minLen) {
+          // append dataframe with contig title information
+          df.push([id, length.toString(), (gc / length).toString(), coverage]);
+        }
+        [id, coverage] = parseContigTitles(line, format);
+        gc = 0;
+        length = 0;
     }
     else {
-      gcCount += calculateLineGC(line);
-      currentContigLength += line.length; 
-      if (df.length != 0 && currentContigLength === Number(length)) { // verifying if current line is last line in contig
-        gc = (gcCount/currentContigLength).toString(); // identified last component of contig title
-        df[df.length - 1][2] = gc; // update gc in dataframe
-      }
+      // adding length and gc count of each line to the total counter for gc and length
+      gc += countGC(line);
+      length += line.length;
     }
   }
 
-  if (df.length === 0) throw 'Error: No contig is bigger than ' + minContigLength + ' base pairs.';
+  if (df.length === 0) throw 'Error: No contig is bigger than ' + minLen + ' base pairs.';
```

Comment:
There is a trick here: After reading the entire file, you need to do this for the last time:

```python
if (id !== NaN && length > minLen) {
  df.push([id, length.toString(), (gc / length).toString(), coverage]);
}
```